examples/approximate_mode.py

```diff
@@ -0,0 +1,188 @@
+import requests
+import tempfile
+from pathlib import Path
+import shutil
+import subprocess
+from time import perf_counter_ns
+
+
+# Video source: https://www.pexels.com/video/dog-eating-854132/
+# License: CC0. Author: Coverr.
+url = "https://videos.pexels.com/video-files/854132/854132-sd_640_360_25fps.mp4"
+response = requests.get(url, headers={"User-Agent": ""})
+if response.status_code != 200:
+    raise RuntimeError(f"Failed to download video. {response.status_code = }.")
+
+temp_dir = tempfile.mkdtemp()
+short_video_path = Path(temp_dir) / "short_video.mp4"
+with open(short_video_path, 'wb') as f:
+    for chunk in response.iter_content():
+        f.write(chunk)
+
+long_video_path = Path(temp_dir) / "long_video.mp4"
+ffmpeg_command = [
+    "ffmpeg",
+    "-stream_loop", "99",  # repeat video 100 times
+    "-i", f"{short_video_path}",
+    "-c", "copy",
+    f"{long_video_path}"
+]
+subprocess.run(ffmpeg_command, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+from torchcodec.decoders import VideoDecoder
+print(f"Short video duration: {VideoDecoder(short_video_path).metadata.duration_seconds} seconds")
+print(f"Long video duration: {VideoDecoder(long_video_path).metadata.duration_seconds / 60} minutes")
+
+# %%
+# .. _perf_creation:
+#
+# Performance: ``VideoDecoder`` creation
+# --------------------------------------
+#
+# In terms of performance, the ``seek_mode`` parameter ultimately affects the
+# **creation** of a :class:`~torchcodec.decoders.VideoDecoder` object. The
+# longer the video, the higher the performance gain.
+
+
+def bench(f, average_over=50, warmup=2, **f_kwargs):
+
+    for _ in range(warmup):
+        f(**f_kwargs)
+
+    times = []
+    for _ in range(average_over):
+        start = perf_counter_ns()
+        f(**f_kwargs)
+        end = perf_counter_ns()
+        times.append(end - start)
+
+    times = torch.tensor(times) * 1e-6  # ns to ms
+    std = times.std().item()
+    med = times.median().item()
+    print(f"{med = :.2f}ms +- {std:.2f}")
+
+
+print("Creating a VideoDecoder object with seek_mode='exact' on a short video:")
+bench(VideoDecoder, source=short_video_path, seek_mode="exact")
+print("Creating a VideoDecoder object with seek_mode='approximate' on a short video:")
+bench(VideoDecoder, source=short_video_path, seek_mode="approximate")
+print()
+print("Creating a VideoDecoder object with seek_mode='exact' on a long video:")
+bench(VideoDecoder, source=long_video_path, seek_mode="exact")
+print("Creating a VideoDecoder object with seek_mode='approximate' on a long video:")
+bench(VideoDecoder, source=long_video_path, seek_mode="approximate")
+
+# %%
+# Performance: frame decoding and clip sampling
+# ---------------------------------------------
+#
+# Strictly speaking the ``seek_mode`` parameter only affects the performance of
+# the :class:`~torchcodec.decoders.VideoDecoder` creation. It does not have a
+# direct effect on frame decoding or sampling. **However**, because frame
```

Comment:
> It does not have a direct effect on frame decoding or sampling.

How about we clarify that we mean effect on performance by adding:

> It does not have a direct effect on the performance of frame decoding or sampling.

I know it's implied by us being in the Performance section, but that tripped me up when I read it.